### PR TITLE
feat: Set up ODD together with python if possible

### PR DESCRIPTION
**Pull request by benjaminhuth**
This modifies the `setup.sh` script a bit so it tries to look for the ODD and add its directory to the `LD_LIBRARY_PATH`.

### CODE REVIEW

**Comment by paulgessinger** [comment]
The ODD repository also has a setup script. Maybe it would be better if this setup checks if that other setup is present, and just sources that?

**Comment by benjaminhuth** [comment]
Hmm this `this_odd.sh` script seems to just add `/usr/local/lib64` (or whatever `cmake` was configured as install prefix I think) to `LD_LIBRARY_PATH`.

The use case of the Acts python `setup.sh` script (at least for me) is for the case ACTS is not installed (which at the moment for the examples also would not work I think).

File: Examples/Python/setup.sh.in [hunk 1]
```diff
@@ -10,17 +10,26 @@
...
     echo "ERROR:   neither ZSH nor Bash was detected, other shells are not supported. The environment has not been modified."
     exit 1
 fi
 
+# Make python package available
+export PYTHONPATH=$python_dir:$PYTHONPATH
+
+# Make ODD available if possible
+odd_dir=$python_dir/../thirdparty/OpenDataDetector/factory
+if [ -d "$odd_dir" ]; then
```
Comment (paulgessinger) [inline Examples/Python/setup.sh.in:28]: `"$OSTYPE" == "darwin"*` for bash I guess, not sure what to do on zsh.

Comment (benjaminhuth) [reply Examples/Python/setup.sh.in:28]: For me both `zsh` and `bash` give `linux-gnu` for `$OSTYPE`, so I would go with that I think

**Comment by Corentin-Allaire** [comment]
Just to comment here on using the ODD with MacOs, it work on my machine.
Maybe doing some thing like this would work here :
` export DYLD_LIBRARY_PATH+=:$odd_dir
export DD4HEP_LIBRARY_PATH+=:$odd_dir`

**Comment by paulgessinger** [comment]
> Just to comment here on using the ODD with MacOs, it work on my machine. Maybe doing some thing like this would work here : ` export DYLD_LIBRARY_PATH+=:$odd_dir export DD4HEP_LIBRARY_PATH+=:$odd_dir`

How do you run then? I have the problem that `DYLD_LIBRARY_PATH` gets cleared and then DD4hep freaks out.

**Comment by Corentin-Allaire** [comment]
> > Just to comment here on using the ODD with MacOs, it work on my machine. Maybe doing some thing like this would work here : ` export DYLD_LIBRARY_PATH+=:$odd_dir export DD4HEP_LIBRARY_PATH+=:$odd_dir`
> 
> How do you run then? I have the problem that `DYLD_LIBRARY_PATH` gets cleared and then DD4hep freaks out.

I just call the python code afterward nothing special. When does it get cleared for you ? (if you want we can discuss this directly via mattermost, might be faster )

File: Examples/Python/setup.sh.in (hunk 1 above)
Comment (Corentin-Allaire) [reply Examples/Python/setup.sh.in:28]: What you could do is : `export LD_LIBRARY_PATH=$odd_dir:$LD_LIBRARY_PATH` in the case of `linux-gnu` and `export DYLD_LIBRARY_PATH+=:$odd_dir export DD4HEP_LIBRARY_PATH+=:$odd_dir` in the case of `darwin` ? That way it is useful to everybody ?

**Review by Corentin-Allaire — APPROVED** [review]
Looks good to me ! I will open one for the macOs side later.

**Pull request merged**